src/System.Text.Json/src/System/Text/Json/Serialization/ReferenceHandlingOnDeserialize.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace System.Text.Json
+{
+    /// <summary>
+    /// This enum defines the various ways the <see cref="Utf8JsonReader"/> can deal with comments.
+    /// </summary>
+    public enum ReferenceHandlingOnDeserialize
```

Comment:
Doesn't this also apply to whether we serialize the metadata?

Reply:
Sorry for not to update the original issue, I was discussing with @ahsonkhan offline about splitting `ReferenceHandling` into two separate properties:

`ReferenceHandlingOnDeserialize` and `ReferenceHandlingOnSerialize` and each one would be used exclusively for deserialize or serialize respectively.